### ISSUE TITLE
Timeseries plot style ideas

### ISSUE THREAD

**Issue by hadinon** [ncsco/pinemap-dss-help#29]
*Change the Projected Change plot to a box and whisker and leave the Occurrence plots as columns? (from Grace)
*Rectangle that covers +/-2 SD and fades out along the edges (Corey will mock up)
*Thicker error bar with no whiskers (Corey will mock up)
*Scatter plot with points (Heather will mock up)
(MEDIUM PRIORITY)

**Comment by hadinon** [comment]
I mocked up these scatterplot time series examples. One has mean, mean +/-1 std dev, and mean +/- 2 std dev. The other only has mean and mean +/-1 std dev.  
![testscatter_timeseriesdss](https://cloud.githubusercontent.com/assets/886682/9693556/50ed0ae0-531f-11e5-878a-3ce6874196e8.png)
![testscatter_timeseriesdss_2](https://cloud.githubusercontent.com/assets/886682/9693557/50ee6016-531f-11e5-95ce-3514a4a6716a.png)

**Comment by daviswx** [comment]
I have generated five different example timeseries plots for the same location (Marshall County, KY).  Thanks for sharing your scatterplot code, which I was easily able to plug in to create those within the DSS interface.

**Column & Whiskers (current style)**
![timeseries_stemwhiskers](https://cloud.githubusercontent.com/assets/9257580/9939683/9160063a-5d39-11e5-8684-d43cbf82a657.png)

**Column & Thick Bars**
![timeseries_thickbars](https://cloud.githubusercontent.com/assets/9257580/9939684/937e6e98-5d39-11e5-94e1-fe098c267c7b.PNG)

**Box & Whiskers**
![timeseries_boxwhiskers](https://cloud.githubusercontent.com/assets/9257580/9959354/513ab1f6-5de1-11e5-815c-aae72e9f43f2.png)

**Faded Bars**
![timeseries_fadedbars](https://cloud.githubusercontent.com/assets/9257580/9940261/bc01ae22-5d3c-11e5-9189-1db01c8fbc3c.png)

**Scatter Range: Mean + 2 Std. Dev.**
![timeseries_scatter2sd](https://cloud.githubusercontent.com/assets/9257580/9939689/98e5a05e-5d39-11e5-8021-72f8f1478b23.png)

**Scatter Range: Mean + 1 & 2 Std. Dev.**
![timeseries_scatterallsd](https://cloud.githubusercontent.com/assets/9257580/9939693/9f2b416c-5d39-11e5-8321-296b32d88629.png)

I guess we need to think about how to evaluate these plots with the beta testers.  Do we ask them to choose their favorite?  Or rank them from best to worst?  Or assign a rating (Very Good, Good, Neutral, Bad, Very Bad) to each?

We could also include space for them the plots and tell us what they like and don't like about them.  And we may want some way of evaluating how they interpret the data on each.

Anything else we need to solicit from them?

**Comment by daviswx** [comment]
I just noticed that the faded bar chart data is a bit different than in the other plots, particularly for RCP4.5.  I will look into it and make sure it's plotting the data correctly.

**Comment by daviswx** [comment]
The Faded Bars plot is updated above.  I was mistakenly pulling from the 1 std. dev. data instead of 2 for RCP 4.5

**Comment by hadinon** [comment]
Ahhh gotcha! Did we also want to include a sixth option which shows the old box and whisker version? 

In terms of evaluating the different options, I like your idea about ranking them because I think it will help us narrow down which one to choose without getting too complex (e.g. assign a rating for each one). And I love the idea of having a spot for them to tell us what they like and don't like. I also would like to add a way to ask about their interpretation, too! All that said, I'm very open to other ideas. Thoughts?

**Comment by daviswx** [comment]
Yes, good idea.  I have added the box and whiskers example as well.

For beta testing, I also like having them rank the options.  That should be easy for them and help us quantitatively evaluate their feedback.

I can think of several "ranking" questions that we might ask:
- Based on how clearly they display future ranges, _rank these timeseries plots from 6 (the best) to 1 (the worst)_.
- Based on how easy it was to determine individual values (e.g., the mean for the 2040-2059 period under current/high emissions levels), _rank these timeseries plots from 6 (the best) to 1 (the worst)_.
- Based on the overall simplicity and success of presenting information, _rank these timeseries plots from 6 (the best) to 1 (the worst)_.

We may want to tweak the wording, but I think this covers most of the things we would want to have them rank.

Here's what I'm thinking for the interpretation question:
- Several of these plots use error bars or something similar to present the future ranges of ±2 standard deviations about the mean.  However, these are more correctly interpreted as confidence intervals, not indicators of model error.  In your opinion, which plots most successfully facilitate this correct interpretation?

I'm also wondering if we should ask whether including the ±1 standard deviation range is helpful.  Here's a possible wording for that question:
- Along with the ±2 standard deviation range, two of these plots (the **Box & Whiskers**, and the **Scatter Range: Mean + 1 & 2 Std. Dev.**) include an additional range showing ±1 standard deviation about the mean.  Did you find it useful to see this range or too much information?

And we could have one more open-ended question asking for any other feedback.

Anything else we need to include?

**Comment by daviswx** [comment]
One more question for the beta testers.  In your opinion, what is the most important piece of information to see about future projections: the mean values or the ranges?

**Comment by hadinon** [comment]
Great questions! I've added all of the above to our beta tester question document.

I have another question idea, which is related to how a user interprets the data. I'm thinking we lead off the question by stating what we want them to take away from these time series graphs. In other words, say, "The intended purpose of this time series graph is to show XYZ." Then, ask them: 
- Were you able to successfully gather that information from at least one of the graphs? If yes, which one? If no, how can we improve?

What do you think?

**Comment by hadinon** [comment]
Also, I made a note that we wanted to ask beta testers about the colors on the time series graphics. Do we still want to do this? I'm pretty happy with the gray colors so I'm inclined to not ask them in hopes that they'd mention it within the open-ended question if they did not like the current colors.

**Comment by daviswx** [comment]
Your question idea sounds good.  I would just change the wording to say "If yes, which ones?" instead of "which one".

I also agree about not asking about the colors.  Like you say, presumably they will tell us if they don't like it. :wink:

**Comment by hadinon** [comment]
Sounds good. Thanks :+1:

**Comment by hadinon** [comment]
Action items based on beta testing: 
*Switch to the Faded Bars plot style. 
*Make it clear (possibly in the expandible FAQ box) that opacity does not represent the weighting of the models.
*Experiment with adding a thin horizontal line on each bar to indicate the location of the mean.
(HIGH PRIORITY)

**Comment by hadinon** [comment]
Additional items from beta testing:
*Darken the darker shade of gray. 
*Duplicate the map title, location (incl. lat/lon), and color ramp/legend (maybe) above the time series. *Mock up some examples of adding an arrow/marker below the column on the time series that matches the currently selected map display.
(MEDIUM PRIORITY)

**Comment by daviswx** [comment]
One more time series suggestion based on today's discussion with Karen McNeal: When the time series loads, flash the title in green (or some other color) to point out that it has loaded.

**Comment by daviswx** [comment]
The faded bars plot style is now in place.  I also added the suggested explanation about the opacity to the FAQ box.

Here's what the bars look like with a horizontal line indicating the mean value:

![timeseries_fadedbars_mean](https://cloud.githubusercontent.com/assets/9257580/11372683/d146ee6a-929c-11e5-90fb-35c52bd93db3.PNG)

Do you prefer this to the non-line version?

**Comment by daviswx** [comment]
Also, I just lightened up the extreme edges of the lighter plot (RCP 4.5) and darkened up the middle of the darker plot (RCP 8.5) to add some more contrast between them:

![timeseries_fadedbars_contrast](https://cloud.githubusercontent.com/assets/9257580/11372806/73170f68-929d-11e5-890e-8b666e568876.png)

**Comment by hadinon** [comment]
Just to recap our discussion, all three of us agree that the faded bars without the horizontal line is the better option at this time. When we add error bars and historical variability, this horizontal line might come back into play.

**Comment by hadinon** [comment]
I just realized this... do we need to update the time series legend to reflect the faded bars? I can't remember if this was mentioned in another issue.

**Comment by daviswx** [comment]
Yes, I still need to update it.  Once I get some of the other structural time series changes implemented, I will work on the minor things like that.

**Comment by daviswx** [comment]
The updated time series title info. is now in place:

![timeseries_newtitle](https://cloud.githubusercontent.com/assets/9257580/11404014/cecc8e66-936d-11e5-8bee-59230808bdfb.PNG)

Question about this: Should this title include the word "Projected" since the time series shows both historical _and_ projected data?  If we removed it, it might say "Average Number of Days Per Year..." or "Change in Average Number of Days Per Year".

One other recent update is reflected on this time series plot: The time slice currently displayed on the maps has its x-axis label bolded.  The map marker is also there, and it's positioned on the side of the emissions scenario displayed on the maps (e.g., in this case, it's on the RCP 8.5 side).  Both of these are quite subtle, so I could experiment with making them more noticeable if needed.

I still need to work on a few more things related to the time series:
- Updating the graphics in the legend
- Updating the y-axis scaling so it's not quite as wide for some plots
- Making sure the colors line up correctly (from #66)

**Comment by hadinon** [comment]
Looks great! I'm fine with saying, "Average..." or "Change in..." but another option might be "Historical and Projected Change/Average..." although that might get lengthy. Thoughts?

I will be honest, I didn't notice the bold x-axis label and map marker until I read your comment. That said, I didn't look very closely at the x-axis so that's probably why! I can't think of a way to make it more noticeable but maybe we can ponder it over turkey. :smiley:

**Comment by daviswx** [comment]
Yeah I think saying both "Historical and Projected" might be too long and wordy.  So I'll just drop that wording for the time series.

And yes, let's revisit the labels next week.  We could also change the text color and/or the font size if we wanted to make them more noticeable, but there may be other good options too.

**Comment by daviswx** [comment]
The updated faded bar images are now in place in the time series legend.  I _think_ that completes all of the time series updates we needed to make, unless there are any other suggestions (like how to better emphasize the x-axis labels).

One question, though: In the legend and tooltips, we're using the word "range", but I seem to remember you saying that "spread" is more appropriate.  So should these instances of "range" be changed to "spread"?

**Comment by hadinon** [comment]
Yes, I would change it to say "spread" instead of "range" (mentioned as a side note in issue #55). 

I like the idea of bumping up the font size of x-axis labels. I'm thinking black is the best color to use in this case. What about putting a border around it? Too complicated?

**Comment by hadinon** [comment]
I just thought of something.... for the legend, I feel like we talked about saying, "Possibilities under reduced/high emissions" instead of saying, "Spread of possibilities..." because that sounded a bit weird. Thoughts? That said, I'm good with changing the tooltip wording to say "spread" instead of "range".

**Comment by daviswx** [comment]
Yes, I like that wording for the legend.  It also saves a bit of space.

That change has been made, along with the updated tooltip wording.

Also, I  increased the x-axis label font size and changed the color to a dark green.  I think this definitely makes them more noticeable.  I also tried adding a border, but IMO it doesn't look as good:
![timeseries_label_border](https://cloud.githubusercontent.com/assets/9257580/11505192/2ee5ecfe-9817-11e5-8b63-d905856828ee.PNG)

Aside from the table (#28) and adding historical variability/error (#51), have all of the necessary changes to the time series been made now?  If so, feel free to close this issue.

**Comment by hadinon** [comment]
I agree about not having a border. Yep, everything seems to be done here. :+1:

**Comment by daviswx** [comment]
Try adding small text below the marker icon saying something like "ON MAP" or "ON MAPS", similar to the "ENLARGE" icon in the map titles.

**Comment by daviswx** [comment]
The new marker icon with "ON MAP" below it is now in place.  What do you think about it?
![timeseries_marker_onmap](https://cloud.githubusercontent.com/assets/9257580/11513548/e0ee11f4-9841-11e5-9746-eca7e9194062.PNG)

Also, I remembered one more thing that needs to be done with the time series.  When the location is loaded, the time series title should flash green (or another color) to indicate that it has loaded.

**Comment by hadinon** [comment]
Yep, that looks awesome. Oh yes, the flash! Would something like this work? http://forums.devshed.com/javascript-development-115/flashing-div-584742.html

**Comment by hadinon** [comment]
I think it is pretty clear now which time period is represented on the maps above but I'm not sure it's immediately obvious which emissions scenario is represented. Maybe we could add a few words to the time series FAQ stating that the "on map" icon will appear below the emissions scenario represented on the maps above? I know there's not much space left in the FAQ so maybe we could reduce some of the other text?

**Comment by daviswx** [comment]
That note has been added to the time series FAQ box.  I didn't reduce any of the other text, but if you have any ideas, I can modify it.

Also, the time series title flashes green once it's loaded.  I just copied the code I used to animate the marker location box.

Other than some of the wording updates mentioned in #31, I think this wraps up the time series changes we wanted to make.

**Comment by hadinon** [comment]
Yep, that all looks good to me.

**Comment by hadinon** [comment]
We went through several iterations of this "ON MAP" icon but at yesterday's meeting, Ryan mentioned mocking something up with the "ON MAP" icon added to the time series plot as an image.

**Comment by daviswx** [comment]
I will investigate this later, but I did find that Highcharts supports [HTML labels](http://api.highcharts.com/highcharts#labels.items.html) on the charts, so we may be able to put that icon closer to the series it describes after all.

**Comment by hadinon** [comment]
Feedback from folks during the Carolinas Climate Resilience Conference tool demo indicates a need for connecting what's being shown on the maps above to the "bar" on the time series plot below. As a supplement (or replacement) of the "ON MAP" icon, we discussed potentially adding vertical lines to highlight which "bar" is being shown on the maps above.

**Comment by hadinon** [comment]
Similarly, at the Carolinas Climate Resilience Conference, we discussed switching the time series plot from horizontal to vertical. This might make it more intuitive that the end values on the bar represent the values on the side maps and that the center value on the bar represents the center map value. By flipping the time series plot to be vertical, the left/right bar values will align with the left/right maps and the center value will align with the center map. 

Related to issue #28, we also talked about putting the time series tabular output next to this vertical time series plot -- another potential benefit of switching the time series to be vertical.
